pint/models/model_builder.py

```diff
@@ -70,8 +70,15 @@ def preprocess_parfile(self, parfile):
         """
         param = {}
         repeat_par = {}
-        pfile = open(parfile, 'r')
-        for l in [pl.strip() for pl in pfile.readlines()]:
+        if isinstance(parfile, str):
+            pfile = open(parfile, 'r')
+            plines = pfile.readlines()
```

Comment:
Do you still need the file object after this? Why not just open it in a with statement and pull out the lines?